lib/closex/caching_client.ex

```
@@ -67,4 +89,17 @@ defmodule Closex.CachingClient do
         {:error, error}
     end
   end
+
+  defp set_cached(key, value) do
+    cache_result = Cachex.set(:closex_cache, key, value)
+
+    case cache_result do
+      {:ok, result} ->
+        Closex.log fn -> "[Closex.CachingClient] SUCCESS setting key: #{key}" end
+        {:ok, Cachex.get(:closex_cache, key)} # Ensure we return the updated resource
+      error ->
```

I'm struggling to understand why this would ever fail?

---

¯\_(ツ)_/¯

It's a good point - given we're not doing anything useful on failure I'll just pattern match on `:ok` and let it raise if there ends up being any error